src/openrct2/ride/TrackData.h

```diff
@@ -67,10 +54,29 @@ constexpr const dodgems_track_size DodgemsTrackSize(track_type_t type)
         return { 4, 4, 59, 123 };
     return { 0, 0, 0, 0 };
 }
-extern const track_type_t TrackElementMirrorMap[TrackElemType::Count];
-
-extern const uint32_t TrackHeightMarkerPositions[TrackElemType::Count];
 
-extern const uint8_t TrackSequenceElementAllowedWallEdges[TrackElemType::Count][16];
+struct TrackElementDescriptor
+{
+    rct_track_coordinates Coordinates;
+
+    rct_preview_track* Block;
+    uint8_t PieceLength;
+    track_curve_chain CurveChain;
+    track_type_t AlternativeType;
+    money32 Pricing;
+    track_type_t MirrorMap;
+    uint32_t HeightMarkerPositions;
+    uint16_t Flags;
+
+    std::array<uint8_t, MaxSequencesPerPiece> SequenceElementAllowedWallEdges;
+    std::array<uint8_t, MaxSequencesPerPiece> TrackSequenceProperties;
+};
 
-extern const uint16_t TrackFlags[TrackElemType::Count];
+namespace OpenRCT2
+{
+    namespace TrackMetaData
+    {
+        void Init();
+        const TrackElementDescriptor& GetTrackElementDescriptor(const uint32_t& type);
```

Comment:
```suggestion
        const TrackElementDescriptor& GetTrackElementDescriptor(const uint32_t type);
```
Trivial types should be passed as value.